doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,416 @@ Compliant example::
        #define _Bool bool
 
 
+Coding Style
+************
+
+General
+=======
+
+GN-01: Each line shall contain at most 120 characters
+-----------------------------------------------------
+
+No more than 120 characters shall be on a line, with tab stops every 8
+characters. Statements longer than this limit shall be broken into multiple
+lines with proper alignment.
+
+Compliant example::
+
+    cpuid(CPUID_EXTEND_FEATURE, &unused,
+        &boot_cpu_data.cpuid_leaves[FEAT_7_0_EBX],
+        &boot_cpu_data.cpuid_leaves[FEAT_7_0_ECX],
+        &boot_cpu_data.cpuid_leaves[FEAT_7_0_EDX]);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       cpuid(CPUID_EXTEND_FEATURE, &unused, &boot_cpu_data.cpuid_leaves[FEAT_7_0_EBX], &boot_cpu_data.cpuid_leaves[FEAT_7_0_ECX], &boot_cpu_data.cpuid_leaves[FEAT_7_0_EDX]);
+
+
+GN-02: Each line shall only contain one statement
+-------------------------------------------------
+
+Compliant example::
+
+    if (condition) {
+        do_a();
+    } else {
+        do_b();
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       if (condition) { do_a();} else {do_b();}
+
+
+GN-03: The semicolon ending a statement shall be the last character of that line
+--------------------------------------------------------------------------------
+
+Compliant example::
+
+    uint32_t a;
+    uint32_t b;
+    uint32_t c;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t a;uint32_t b;uint32_t c;
+
+
+
+Indentation
+===========
+
+IT-01: Tabs shall be used for code indentation
+----------------------------------------------
+
+Spaces are allowed only for indenting comments or aligning statements that span
+multiple lines.
+
+Compliant example::
+
+    if (condition) {
+        do_a();
+    } else {
+        do_b();
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       if (condition) {
+         do_a();
+       } else {
+         do_b();
+       }
+
+
+IT-02: Tabs shall be 8 characters wide
+--------------------------------------
+
+A tab character shall be considered 8-character wide when limiting the line
+width.
+
+Compliant example::
+
+    /* This rule is straightforward. Example is not needed. */
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* This rule is straightforward. Example is not needed. */
+
+
+
+Whitespaces
+===========
+
+WS-01: Trailing whitespaces shall not be allowed at the end of lines
+--------------------------------------------------------------------
+
```

Comment:
You could mention spaces **and tabs** are not allowed at the end of a line.

Reply:
Will update it in v4.